internal/app/cache/cache_test.go

```diff
@@ -0,0 +1,136 @@
+package cache
+
+import (
+	"testing"
+	"time"
+
+	envoy_api_v2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/stretchr/testify/assert"
+)
+
+const testKeyA = "key_A"
+const testKeyB = "key_B"
+
+func onEvict(key uint64, conflict uint64, value interface{}, cost int64) {
+	// TODO: Simulate eviction behavior, e.g. closing of streams.
+}
+
+var testRequest = envoy_api_v2.DiscoveryRequest{
+	VersionInfo: "version_A",
+	Node: &core.Node{
+		Id:      "id_A",
+		Cluster: "cluster_A",
+	},
+	ResourceNames: []string{"resource_A"},
+	TypeUrl:       "typeURL_A",
+	ResponseNonce: "nonce_A",
+}
+
+var testResponse = envoy_api_v2.DiscoveryResponse{
+	VersionInfo: "version_A",
+	Resources:   []*any.Any{},
+	Canary:      false,
+	TypeUrl:     "typeURL_A",
+	Nonce:       "nonce_A",
+	ControlPlane: &core.ControlPlane{
+		Identifier: "identifier_A",
+	},
+}
+
+func TestExists_EmptyCache(t *testing.T) {
+	cache, err := NewCache(10, 1048576, time.Second*60, onEvict)
```

Comment:
I was thinking more about this..Do we use two different testing patterns or just the BDD pattern everywhere?
@jessicayuen @eapolinario @LisaLudique What is your opinion?

Reply:
I don't think we should be married to BDD necessarily, i.e. use it when it makes sense, otherwise, regular xUnit-type tests can (and should) be used.

Reply:
sounds good